app/controllers/graders_controller.rb

```diff
@@ -98,21 +98,27 @@ def global_actions
 
       case params[:global_actions]
       when 'assign'
-        if params[:skip_empty_submissions] == 'true'
-          # If the instructor wants to skip empty submissions, filter
-          # groups with empty submissions into a new array
-          filtered_grouping_ids = filter_empty_submissions(grouping_ids)
-          if filtered_grouping_ids.count != grouping_ids.count
-            found_empty_submission = true
+        begin
```

Comment:
Overall the logic (for detecting inactive TAs) is correct, but it would be better to pull it out of the helper and instead put it at the start of the controller method, in the same block as Lines 80-88 that are doing the initialization of `grader_ids`. This will cause the filtering to apply right at the start. Enable the inactive-grader filter whenever an "assign" or "randomly assign" action is selected, but _not_ an "unassign". That is, we'll allow inactive TAs to be unassigned from existing groups/criteria, but not assigned to new ones.

Also if you move this check up there, you can use `flash_message` directly rather than raising an error.